project/tests/com/google/daggerquery/executor/services/DaggerQueryControllerTest.java

```diff
@@ -0,0 +1,129 @@
+package com.google.daggerquery.executor.services;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import com.google.daggerquery.protobuf.autogen.BindingGraphProto;
+import com.google.daggerquery.protobuf.autogen.DependencyProto;
+import org.junit.Test;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+
+public class DaggerQueryControllerTest {
```

Comment:
It appears to me this test largely overlaps with `QueryExecutorTest`, and tests the exact output in addition. How about just keeping one of them to reduce the maintenance overhead?

Reply:
Yes, there are a big intersection in cases which are tested by these two classes. The main purpose of the `QueryExecutorTest` implementation was to test the actual behavior of a program when a user uses it, but it was mostly testing an output, which isn't good. I removed it.